plugins/bulk_upload/csv/batch/BulkUploadEngineCsv.php

```diff
@@ -157,27 +157,48 @@ protected function addBulkUploadResult(KalturaBulkUploadResult $bulkUploadResult
 	 */
 	protected function createUploadResult($values, $columns)
 	{
-	    if($this->handledRecordsThisRun > $this->maxRecordsEachRun)
+		if($this->handledRecordsThisRun > $this->maxRecordsEachRun)
 		{
 			$this->exceededMaxRecordsEachRun = true;
 			return null;
 		}
 		$this->handledRecordsThisRun++;
-		
-	    $bulkUploadResult = $this->getUploadResultInstance();
+
+		$foundUnknownEncoding = false;
+
+		$bulkUploadResult = $this->getUploadResultInstance();
+		$bulkUploadResult->bulkUploadResultObjectType = $this->getUploadResultInstanceType();
 		$bulkUploadResult->bulkUploadJobId = $this->job->id;
 		$bulkUploadResult->lineIndex = $this->lineNumber;
 		$bulkUploadResult->partnerId = $this->job->partnerId;
 		//CSV files allow values to contain the "," character, on the condition that the value is surrounded by "".
 		for ($index = 0; $index < count($values); $index++)
 		{
-		    if (strpos($values[$index], ",") !== false)
-		    {
-		        $values[$index] = '"'.$values[$index].'"';
-		    }
+			if (strpos($values[$index], ",") !== false)
+			{
+				$values[$index] = '"'.$values[$index].'"';
+			}
+
+			if(!mb_detect_encoding($values[$index]))
+			{
+				KalturaLog::warning('encoding cannot be detected');
+				$values[$index] = 'encoding cannot be detected';
+				$foundUnknownEncoding = true;
+			}
 		}
 		$bulkUploadResult->rowData = implode(",", $values);
-		
+
+		if ($foundUnknownEncoding)
+		{
+			$bulkUploadResult->objectStatus = KalturaUserStatus::ACTIVE;
+			$bulkUploadResult->status = KalturaBulkUploadResultStatus::ERROR;
+			$bulkUploadResult->errorType = KalturaBatchJobErrorTypes::APP;
+			$bulkUploadResult->errorDescription = 'value with unknown encoding was passed';
+
+			$this->addBulkUploadResult($bulkUploadResult);
+			return null;
```

Comment:
Should we return null or the defected results?

Reply:
null.
we're dealing with it one line before, by 'addBulkUploadResult'.
then returning null, same if we reached 'maxRecordsEachRun' at the beginning of this function.